### DISCUSSION TITLE
Pinterest - boards not populating.

### DISCUSSION THREAD

**Discussion by nashvegasnate** [nss-evening-cohort-14/evening-client-side#173] [Help Me!] [answered]
Nathan Keith

Have you searched the other help tickets? If not, do that first (Duplicates asks for help will be shamed ceaselessly) 
YES

If there are similar issues, make sure to give us their number/link to them and tell how yours is different

Describe your issue
I have data and rules set up in my firebase. Upon login, my log out button disappears and board data doesn't render. I have a navigation events file and not sure it is doing what I expected it to do.

List any error messages that appear
Error innerHTML of null on the logout button container.

What have you googled? (Paste links to the things you have found from your google search - You must include at least 3 links)
Nothing yet. Not sure what is happening.

What have you tried? Tell us everything you have tried.

Copy/paste link TO THE FILE on your branch or in your repo that you are having issues with NO SCREEN SHOTS 
REPO: https://github.com/nashvegasnate/ASSIGNMENT-Firebase-Pinterest
BRANCH: https://github.com/nashvegasnate/ASSIGNMENT-Firebase-Pinterest/tree/boardBuild
NAVIGATION EVENTS: https://github.com/nashvegasnate/ASSIGNMENT-Firebase-Pinterest/blob/boardBuild/src/javascripts/events/navigationEvents.js
HOME: https://github.com/nashvegasnate/ASSIGNMENT-Firebase-Pinterest/blob/boardBuild/src/javascripts/components/home.js
NAVBAR: https://github.com/nashvegasnate/ASSIGNMENT-Firebase-Pinterest/blob/boardBuild/src/javascripts/components/navBar.js

ERROR:
selectBoard.js?362a:21 Uncaught (in promise) TypeError: Cannot set property 'innerHTML' of null
    at eval (selectBoard.js?362a:21)
eval @ selectBoard.js?362a:21
Promise.then (async)
selectBoard @ selectBoard.js?362a:10
addBoardForm @ addBoardForm.js?7038:18
eval @ domEvents.js?5713:78

:smiling_face_with_3_hearts:
NOTE: We will not accept links to the whole repo. You must provide a link directly to the specific file

**Comment by nashvegasnate** [comment]
To elaborate on my issue, this information might be helpful.
My boards are not popping up on the DOM. I am logged in and there is sample data connected to my uid, however, for some reason nothing is happening. I have corrected my "innerHTML of null" problem on the initial app load, yet still no boards. My logout button has also disappeared and I believe there is some connection (or disconnection) with my domBuilder that has brought that on.

**Comment by drteresavasquez** [comment]
There are several tickets for innerHTML of null. Here is one: https://github.com/nss-evening-cohort-14/evening-client-side/discussions/160

We need to see the error so please copy/paste it in the ticket along with the links to your code and what you have tried.

**Comment by drteresavasquez** [comment]
There is a lot of code that seems to be targeting the same divs. Make sure you do a workflow like we did in class to identify each portion of your code and the files. You need to start from the beginning of the flow and see what is targeting what and at what time things should load. 

Your error is telling you that you don't have an element on the DOM to select. If you've cleared that up and no boards, that may mean your not targeting the element where the board should render.

**Reply by nashvegasnate** [reply]
So, it looks like I was able to clear up the login/logout button issue. I don't remember doing it, but I had evidently created separate divs to house those buttons, so they ended up hidden on the DOM because they weren't properly targeted. Simple fix after hours of headache!
Despite fixing the above issue, my boards are not yet populating, but I am working to get that corrected at the moment.

**Reply by drteresavasquez** [reply]
Keep us posted on progress. If you take the advice above you should be able to read the boards.

**Reply by nashvegasnate** [reply]
Boards are now being read! My personal uid was manually entered into the json data after it had been added to the file, so it wasn't making the connection to Firebase. I had to re-import my json data into Firebase and the boards popped up! I'm gonna have to give some credit to Tad @tsekeres for coming up with that solution!

**Reply by drteresavasquez** [reply]
Post your answer in the main thread so I can mark it as the answer. Good job!

**Comment by nashvegasnate** [comment, marked as answer]
Boards are now being read! My personal uid was manually entered into the json data after it had been added to the file, so it wasn't making the connection to Firebase. I had to re-import my json data into Firebase and the boards popped up! I'm gonna have to give some credit to Tad @tsekeres for coming up with that solution!
